### PR TITLE
feat(database): faster queries from the persons table

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

The existing `select * from persons` transformation into `select * from (select ... from raw_persons)` is inefficient and times out for large customers, even on the new cluster.

## Changes

Swaps it out for a faster version:

```sql
SELECT * FROM raw_persons WHERE (id, version) IN (
   SELECT id, max(version) as version
   FROM raw_persons
   GROUP BY id
   HAVING ifNull(equals(argMax(raw_persons.is_deleted, raw_persons.version), 0), 0)
)
```

## How did you test this code?

Tested the query in the SQL editor. Hoping CI will update the snapshots and report of failures.

### CODE REVIEW

File: posthog/hogql/database/schema/persons.py [hunk 1]
```diff
@@ -30,14 +29,30 @@
...
-        argmax_field="version",
-        deleted_field="is_deleted",
+    from posthog.hogql.parser import parse_select
+    from posthog.hogql import ast
+
+    query = parse_select(
+        """
+        SELECT id FROM raw_persons WHERE (id, version) IN (
+           SELECT id, max(version) as version
+           FROM raw_persons
```
Comment (timgl) [inline posthog/hogql/database/schema/persons.py:39]: This should be filtered by team id

File: posthog/hogql/database/schema/persons.py [hunk 2]
```diff
@@ -30,14 +29,30 @@
...
-        select_fields=requested_fields,
-        group_fields=["id"],
-        argmax_field="version",
-        deleted_field="is_deleted",
+    from posthog.hogql.parser import parse_select
+    from posthog.hogql import ast
+
+    query = parse_select(
+        """
+        SELECT id FROM raw_persons WHERE (id, version) IN (
```
Comment (timgl) [inline posthog/hogql/database/schema/persons.py:37]: What happens with this id? Does it get replaced with the actual fields we need (properties, created_at) or is this going into another subquery?

File: posthog/hogql/database/schema/persons.py (hunk 1 above)
Comment (mariusandra) [reply posthog/hogql/database/schema/persons.py:39]: it's automatically

File: posthog/hogql/database/schema/persons.py (hunk 2 above)
Comment (mariusandra) [reply posthog/hogql/database/schema/persons.py:37]: look a few lines lower

**Review by timgl — APPROVED** [review]

**Comment by mariusandra** [comment]
Ran some benchmarks... and I'm not sure anymore about anything.

Running the old `select * from persons` query against the new one for our team... while also selecting `properties` (blue is old, yellow is new):
image

However if I exclude `properties` from the queries, the picture is drastically different (blue is old, yellow is new)

image

Even in metabase, the old query returns almost instantly, whereas the new one takes many seconds.

🤔

**Comment by mariusandra** [comment]
I reworked this optimization to be a Query Modifier that can be toggled on and off as needed.



It currently defaults to off. With this in, we can now run test the query for a whole host of customers easier.

**Pull request merged**